### PR TITLE
Add vehicle license plate prototype

### PR DESCRIPTION
**Pull request by Thurston00**
Add license plates as part of Vehicle Ownership (see #36).

To add a license plate, follow the steps below:

1. Override the vehicle prefab that you want to add the license plate to.
2. Open the overridden prefab.
3. Inside of the `Object Properties`, search for `EL_LicensePlateManagerComponent`.
4. Add a plate by clicking the `+` button: 
image

5. Drop down the create entity, and click `set class` for the `Point` entry.
image

6. To position your plate, click `Offset` and put it into the desired position.
7. Click `Apply to prefab` and then spawn the vehicle, and the license plate will be applied.

Preview of the license plates:
image

### CODE REVIEW

**Comment by wrdg** [comment]
lgtm

File: src/Scripts/Game/Components/Vehicle/EL_LicensePlateManagerComponent.c [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+		SetEventMask(owner, EntityEvent.INIT);
+		owner.SetFlags(EntityFlags.ACTIVE, false);
+    }
+	
+    bool GetPositionFromPoint(int index, out vector transform[4])
+    {
+        Math3D.MatrixIdentity4(transform);
+        BaseContainer source;
+        GenericEntity entity = GetOwner();
+		BaseContainerList list;
```
[the remark this answers is not in the document]
Comment (thebonbon) [reply src/Scripts/Game/Components/Vehicle/EL_LicensePlateManagerComponent.c:97]: same for 3 more further down

**Review by thebonbon — CHANGES_REQUESTED** [review]

File: src/Scripts/Game/Components/Vehicle/EL_LicensePlateManagerComponent.c [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    protected ResourceName m_LicensePlatePrefab;
+	
+	[RplProp(onRplName: "OnRegistrationUpdated")]
+	string m_Registration;
+	
+	override void EOnInit(IEntity owner)
+	{
+		Resource resource = Resource.Load(m_LicensePlatePrefab);
+		
+		if (!resource.IsValid())
```
[the remark this answers is not in the document]
Comment (thebonbon) [reply src/Scripts/Game/Components/Vehicle/EL_LicensePlateManagerComponent.c:42]: Also Check [Code Styleguide](https://github.com/EveronLife/EveronLife/blob/main/.github/CONTRIBUTING.md#-code-styleguide) for sorting methods

**Comment by Thurston00** [comment]
Good eye @thebonbon, I had forgotten to check for that, I usually have a formatter in my normal IDE.

File: src/Scripts/Game/Components/Vehicle/EL_LicensePlateManagerComponent.c [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+	void OnRegistrationUpdated()
+	{
+		foreach (auto plate : m_Plates)
+		{
+			plate.m_Object.m_TextWidget.SetText(m_Registration);
+		}
+	}
+	
+	override void OnPostInit(IEntity owner)
+    {
```
Comment (thebonbon) [inline src/Scripts/Game/Components/Vehicle/EL_LicensePlateManagerComponent.c:86]: Indent

File: src/Scripts/Game/Vehicle/EL_LicensePlateEntity.c [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+			if (!m_wRenderTargetTextureWidget)
+			{
+				m_wPIPRoot.RemoveFromHierarchy();
+				m_wPIPRoot = null;
+				return;
+			}
+			
+			m_TextWidget = TextWidget.Cast(m_wRenderTargetTextureWidget.FindAnyWidget("Text0"));
+			
+			if (m_LicensePlateManager && m_TextWidget)
```
Comment (thebonbon) [inline src/Scripts/Game/Vehicle/EL_LicensePlateEntity.c:58]: Single line if

File: src/Scripts/Game/Components/Vehicle/EL_LicensePlateManagerComponent.c [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+class EL_LicensePlateComponent
+{
+	[Attribute(uiwidget: UIWidgets.Auto)]
+	ref EL_PointInfo m_Point;
+	
+	EL_LicensePlateEntity m_Object;
+	
+	void ~EL_LicensePlateComponent()
+	{
+		if (m_Object)
```
Comment (thebonbon) [inline src/Scripts/Game/Components/Vehicle/EL_LicensePlateManagerComponent.c:16]: Single line if

**Review by thebonbon — CHANGES_REQUESTED** [review]
Couple more. Other than that, looks good!

File: src/Scripts/Game/Components/Vehicle/EL_LicensePlateManagerComponent.c (hunk 5 above)
Comment (Arkensor) [reply src/Scripts/Game/Components/Vehicle/EL_LicensePlateManagerComponent.c:16]: Those should be fine. Or is there an official guideline against it?

Comment (thebonbon) [reply src/Scripts/Game/Components/Vehicle/EL_LicensePlateManagerComponent.c:16]: No guideline but should be consistent right?

File: src/Scripts/Game/Components/Vehicle/EL_LicensePlateManagerComponent.c [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+		
+		if (!resource.IsValid()) return;
+		
+		EntitySpawnParams params();
+		params.Parent = owner;
+		params.TransformMode = ETransformMode.LOCAL;
+		
+		RplComponent rpl = RplComponent.Cast(owner.FindComponent(RplComponent));
+		
+		if (rpl && rpl.IsMaster())
```
Comment (Jacob-Mango) [inline src/Scripts/Game/Components/Vehicle/EL_LicensePlateManagerComponent.c:50]: This should be swapped so the RplComponent is checked only when the game is playing and not inside the World Editor window in Edit Mode.

File: src/Scripts/Game/Components/Vehicle/EL_LicensePlateManagerComponent.c [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+			{
+				Resource container = BaseContainerTools.LoadContainer("{B1DD7B5D4812AB19}Configs/Vehicles/VehicleSettings.conf");
+				EL_VehicleSettings vehicleSettings = EL_VehicleSettings.Cast(BaseContainerTools.CreateInstanceFromContainer(container.GetResource().ToBaseContainer()));
+				m_Registration = vehicleSettings.m_LicensePlateGenerator.GenerateLicensePlate();
+			}
+			else
+			{
+				m_Registration = "PLACEHOLDER";
+			}
+		}
```
Comment (Arkensor) [inline src/Scripts/Game/Components/Vehicle/EL_LicensePlateManagerComponent.c:62]: Optimize the check for license plate generation
```suggestion
		if (GetGame().InPlayMode() && rpl && rpl.IsMaster())
		{
			Resource container = BaseContainerTools.LoadContainer("{B1DD7B5D4812AB19}Configs/Vehicles/VehicleSettings.conf");
			EL_VehicleSettings vehicleSettings = EL_VehicleSettings.Cast(BaseContainerTools.CreateInstanceFromContainer(container.GetResource().ToBaseContainer()));
			m_Registration = vehicleSettings.m_LicensePlateGenerator.GenerateLicensePlate();
		}
```

File: src/Scripts/Game/Components/Vehicle/EL_LicensePlateManagerComponent.c [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+		if (m_Object)
+		{
+			delete m_Object;
+		}
+	}
+}
+
+class EL_PointInfo: PointInfo
+{
+}
```
Comment (Arkensor) [inline src/Scripts/Game/Components/Vehicle/EL_LicensePlateManagerComponent.c:25]: Move to a seperate file

**Review by Arkensor — APPROVED** [review]

**Pull request merged**